presto-hive/src/main/java/com/facebook/presto/hive/HiveZeroRowFileCreator.java

```diff
@@ -144,10 +149,14 @@ public void createFiles(ConnectorSession session, HdfsContext hdfsContext, Path
         }
     }
 
-    private void createFile(HdfsContext hdfsContext, Path path, byte[] content)
+    private void createFile(HdfsContext hdfsContext, Path path, byte[] content, ConnectorSession session)
     {
-        try (OutputStream outputStream = hdfsEnvironment.getFileSystem(hdfsContext, path).create(path, false)) {
-            outputStream.write(content);
+        try {
+            FileSystem fs = hdfsEnvironment.getFileSystem(hdfsContext, path);
+            try (DataSink dataSink = dataSinkFactory.createDataSink(session, fs, path)) {
```

Comment:
After switching to `datasink`,  we would have retry capability? I am trying to understand what is the issue of using `OutputStream`